### ISSUE TITLE
Unable to get cadvisor running for monitoring docker containers - files missing error

### ISSUE THREAD

**Issue by jmat6** [google/cadvisor#1494]
Hi

I have a 3 master, 5 data node Elastic search test cluster. On a separate machine, I have a docker container running. Now, I want to send the host metrics (on which docker application is running) and the docker metrics to the EL cluster. I have Kibana set up over the elastic search cluster, so I don't want to/need to monitor the data from cadvisor UI, hence I omitted the publish flag in the following command: 

`sudo docker run --volume=/:/rootfs:ro --volume=/var/run:/var/run:rw --volume=/sys:/sys:ro --volume=/var/lib/docker/:/var/lib/docker:ro --privileged=true --volume=/cgroup:/cgroup:ro --detach=true  google/cadvisor:latest -storage_driver="elasticsearch" -alsologtostderr=true -storage_driver_es_host="http://10.25.11.133:9200"`

However, the docker container immediately exists with the following logs: 

```
I1007 09:24:23.736924       1 storagedriver.go:48] Using backend storage type "elasticsearch"
I1007 09:24:23.737060       1 storagedriver.go:50] Caching stats in memory for 2m0s
I1007 09:24:23.737135       1 manager.go:140] cAdvisor running in container: "/docker/66ae6e432f7201494e09607184f950450f50d87ee4689c0b50bf31e293eada30"
W1007 09:24:23.746538       1 manager.go:148] unable to connect to Rkt api service: rkt: cannot tcp Dial rkt api service: dial tcp [::1]:15441: getsockopt: connection refused
I1007 09:24:23.756258       1 fs.go:116] Filesystem partitions: map[/dev/mapper/docker-202:1-129297-635ed0921ff2f7ecbc6a0b169d329e7addd07f54da7586ffbfdf178ea9a8b4ef:{mountpoint:/ major:253 minor:2 fsType:xfs blockSize:0} /dev/xvda1:{mountpoint:/var/lib/docker/devicemapper major:202 minor:1 fsType:xfs blockSize:0} /dev/mapper/docker-202:1-129297-b674f7a1f8238f9e89f3f45f45c3103cd2c1c294b8824733c331f32180f33a86:{mountpoint:/rootfs/var/lib/docker/devicemapper/mnt/b674f7a1f8238f9e89f3f45f45c3103cd2c1c294b8824733c331f32180f33a86 major:253 minor:1 fsType:xfs blockSize:0}]
I1007 09:24:23.759935       1 manager.go:195] Machine: {NumCores:1 CpuFrequency:2400066 MemoryCapacity:1039843328 MachineID:f32e0af35637b5dfcbedcb0a1de8dca1 SystemUUID:EC290DDB-B3C7-116E-83F5-27C0DDF0E749 BootID:0e8dd135-3de3-453c-bafb-2d4b9ec04af9 Filesystems:[{Device:/dev/mapper/docker-202:1-129297-635ed0921ff2f7ecbc6a0b169d329e7addd07f54da7586ffbfdf178ea9a8b4ef Capacity:10725883904 Type:vfs Inodes:10484736 HasInodes:true} {Device:/dev/xvda1 Capacity:8578400256 Type:vfs Inodes:8387584 HasInodes:true} {Device:/dev/mapper/docker-202:1-129297-b674f7a1f8238f9e89f3f45f45c3103cd2c1c294b8824733c331f32180f33a86 Capacity:10725883904 Type:vfs Inodes:10484736 HasInodes:true}] DiskMap:map[253:1:{Name:dm-1 Major:253 Minor:1 Size:10737418240 Scheduler:none} 253:2:{Name:dm-2 Major:253 Minor:2 Size:10737418240 Scheduler:none} 202:0:{Name:xvda Major:202 Minor:0 Size:8589934592 Scheduler:deadline} 253:0:{Name:dm-0 Major:253 Minor:0 Size:107374182400 Scheduler:none}] NetworkDevices:[{Name:eth0 MacAddress:0e:64:2d:c2:01:3d Speed:0 Mtu:9001}] Topology:[{Id:0 Memory:1073336320 Cores:[{Id:0 Threads:[0] Caches:[{Size:32768 Type:Data Level:1} {Size:32768 Type:Instruction Level:1} {Size:262144 Type:Unified Level:2}]}] Caches:[{Size:31457280 Type:Unified Level:3}]}] CloudProvider:AWS InstanceType:t2.micro InstanceID:i-0a6963ca9b9364a80}
I1007 09:24:23.760469       1 manager.go:201] Version: {KernelVersion:3.10.0-327.13.1.el7.x86_64 ContainerOsVersion:Alpine Linux v3.4 DockerVersion:1.12.1 CadvisorVersion:v0.24.0 CadvisorRevision:0cdf491}
E1007 09:24:23.766147       1 factory.go:291] devicemapper filesystem stats will not be reported: unable to find thin_ls binary
I1007 09:24:23.766157       1 factory.go:295] Registering Docker factory
W1007 09:24:23.766168       1 manager.go:244] Registration of the rkt container factory failed: unable to communicate with Rkt api service: rkt: cannot tcp Dial rkt api service: dial tcp [::1]:15441: getsockopt: connection refused
I1007 09:24:23.766172       1 factory.go:54] Registering systemd factory
I1007 09:24:23.766285       1 factory.go:86] Registering Raw factory
I1007 09:24:23.766380       1 manager.go:1082] Started watching for new ooms in manager
W1007 09:24:23.766664       1 manager.go:272] Could not configure a source for OOM detection, disabling OOM events: unable to find any kernel log file available from our set: [/var/log/kern.log /var/log/messages /var/log/syslog]
I1007 09:24:23.766983       1 manager.go:285] Starting recovery of all containers
I1007 09:24:23.819583       1 manager.go:290] Recovery completed
F1007 09:24:23.826940       1 cadvisor.go:151] Failed to start container manager: inotify_add_watch /sys/fs/cgroup/cpuacct,cpu: no such file or directory
Elasticsearch returned with code 200 and version 2.4.1

```

Am I missing something here? The `-storage_driver_es_host="http://10.25.11.133:9200"` specifies the URL of the machine running the Elastic search data node. 

Also, I am not clear how can I send the host machine metrics to the cluster?

**Comment by derekwaynecarr** [comment]
This should have been fixed.  I will try to reproduce.

**Comment by jbh6** [comment]
@derekwaynecarr , any updates on this?

**Comment by BrickXu** [comment]
join this thread. 

hi @derekwaynecarr , could you please give the original issue refs ?

**Comment by BrickXu** [comment]
I think this PR would fix it #1476

**Comment by derekwaynecarr** [comment]
This was fixed, but a new release has not yet been pushed to docker hub that contains the fix.

For more detail, see here:
https://github.com/google/cadvisor/pull/1476#issuecomment-252070142

**Comment by mpas** [comment]
I seem to still keep hitting the issue.. this with release 0.24.1

`cadvisor.go:151] Failed to start container manager: inotify_add_watch /var/lib/docker/devicemapper/mnt/ed5946c7c034f2cb5bd752e1b2bc1856ba9155d27e3df8a3fdad4fdbd9694ebb/rootfs/rootfs/var/lib/docker/devicemapper/mnt/ed5946c7c034f2cb5bd752e1b2bc1856ba9155d27e3df8a3fdad4fdbd9694ebb/rootfs/sys/fs/cgroup/cpu: no such file or directory`

**Comment by rajivgandhi1** [comment]
The fix doesnt work for me tried both 0.24.1 and 0.25.0

**Comment by l15k4** [comment]
I've got the same problem on : 
```
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.2 LTS
Release:	16.04
Codename:	xenial
```
Identical log messages as in this issue. I'm using the latest `google/cadvisor:latest`, `privileged` mode doesn't help. I run it as `root`...
